doc/source/tune-config.rst

```diff
@@ -0,0 +1,78 @@
+
+    # Single experiment
+    run_experiments(Experiment(...))
+
+    # Multiple experiments
+    run_experiments([Experiment(...), Experiment(...), ...])
+
+.. autoclass:: ray.tune.Experiment
+
+An example of this can be found in `hyperband_example.py <https://github.com/ray-project/ray/blob/master/python/ray/tune/examples/hyperband_example.py>`__.
+
+Alternatively, you can pass in a JSON object. This uses the same fields as
+the `ray.tune.Experiment`, except the experiment name is the key of the top level
+dictionary.
+
+.. code-block:: python
+
+    run_experiments({
+        "my_experiment_name": {
+            "run": "my_func",
+            "resources": { "cpu": 1, "gpu": 0 },
+            "stop": { "mean_accuracy": 100 },
+            "config": {
+                "alpha": grid_search([0.2, 0.4, 0.6]),
+                "beta": grid_search([1, 2]),
+            },
+            "upload_dir": "s3://your_bucket/path",
+            "local_dir": "~/ray_results",
+            "max_failures": 2
+        }
+    })
+
+An example of this can be found in `async_hyperband_example.py <https://github.com/ray-project/ray/blob/master/python/ray/tune/examples/async_hyperband_example.py>`__.
+
+
+Trial Variant Generation
+------------------------
+
+In the above example, we specified a grid search over two parameters using the ``grid_search`` helper function. Ray Tune also supports sampling parameters from user-specified lambda functions, which can be used in combination with grid search.
+
+The following shows grid search over two nested parameters combined with random sampling from two lambda functions. Note that the value of ``beta`` depends on the value of ``alpha``, which is represented by referencing ``spec.config.alpha`` in the lambda function. This lets you specify conditional parameter distributions.
+
+.. code-block:: python
+
+    "config": {
+        "alpha": lambda spec: np.random.uniform(100),
+        "beta": lambda spec: spec.config.alpha * np.random.normal(),
+        "nn_layers": [
+            grid_search([16, 64, 256]),
+            grid_search([16, 64, 256]),
+        ],
+    },
+    "repeat": 10,
+
+By default, each random variable and grid search point is sampled once. To take multiple random samples or repeat grid search runs, add ``repeat: N`` to the experiment config. E.g. in the above, ``"repeat": 10`` repeats the 3x3 grid search 10 times, for a total of 90 trials, each with randomly sampled values of ``alpha`` and ``beta``.
+
+For more information on variant generation, see `variant_generator.py <https://github.com/ray-project/ray/blob/master/python/ray/tune/variant_generator.py>`__.
+
+
+Resource Allocation
+-------------------
+
+Ray Tune runs each trial as a Ray actor, allocating the specified GPU and CPU ``resources`` to each actor (defaulting to 1 CPU per trial). A trial will not be scheduled unless at least that amount of resources is available in the cluster, preventing the cluster from being overloaded.
```

Comment:
Add a note that if GPU resources are not requested, CUDA_VISIBLE_DEVICES will be set to the empty script, disallowing GPU access.

Reply:
done